api/src/certification/session/domain/models/ComplementaryCertification.js

```diff
@@ -1,7 +1,25 @@
-export class ComplementaryCertification {
+import { ComplementaryCertificationKeys } from '../../../shared/domain/models/ComplementaryCertificationKeys.js';
+
+class ComplementaryCertification {
+  /**
+   * @param {Object} props
+   * @param {number} props.id
+   * @param {string} props.label
+   * @param {ComplementaryCertificationKeys} props.key identifier key
+   */
   constructor({ id, label, key } = {}) {
+    ComplementaryCertification.#assertKey(key);
```

Comment:
C'est propre ici, mais est-ce que les deux indirections pour un includes sont pas un peu coûteuses à la relecture ?
```suggestion
Object.values(COMPLEMENTARY_CERTIFICATION_KEYS).includes(key) || throw new TypeError('Illegal argument provided');
```

Reply:
`Object.values(COMPLEMENTARY_CERTIFICATION_KEYS) `n'est pas possible (je décris + bas pourquoi)

Si je fais
``
`> Object.values(ComplementaryCertificationKeys)
[
  'DROIT',
  'CLEA',
  'EDU_1ER_DEGRE',
  'EDU_2ND_DEGRE',
  [Function: contains]
]
```

Et je trouve pas propre le "[Function: contains]" de ce Object.values (et ça sera encore pire si l'objet vient commencer à prendre plusieurs fonctions).

Pour être honnête je trouve que JS a encore beaucoup à faire sur les enum. Mais c'est pas un language connu pour sa protection des objets, alors un objet verrouillé comme un enum c'est + que du sucre syntaxique comme apporte `class` 😅 

C'est en fait là toute la raison de ce code, d'avoir un objet "freeze" pour éviter les modifications venues de n'importe où (cet enum est quand même une pierre angulaire de notre appli) et par contre de ne pas avoir des objets Domain anémiques (donc pouvoir rajouter des fonctions comme ici le `contains`).

Par conséquent pour le moment je préfère l'implémentation actuelle 😊